Scripts/utils/print_links.py

```diff
@@ -1,6 +1,57 @@
+from typing import Iterable, Union, Tuple
+from shapely import Point, LineString
+
 from utils.calc_noise import NoiseModel
 
 
+class NODE:
+    geom_type = "Point"
+
+    def __new__(cls, node):
+        return Point(node.x, node.y)
+
+
+class LINK:
+    geom_type = "LineString"
+
+    def __new__(cls, link):
+        return LineString(link.shape)
+
+
+def geometries(scenario,
+               objects: Iterable,
+               geom_type: Union[NODE, LINK]) -> Tuple[Iterable, dict]:
+    """Turn EMME network objects into GeoJSON records.
+
+    Parameters
+    ----------
+    scenario : inro.emme.database.scenario.Scenario
+        Scenario for which extra attributes are exported
+    objects : Iterable
+        Iterator over network objects (links or nodes)
+    geom_type : NODE or LINK
+        NODE or LINK geometry type
+
+    Returns
+    -------
+    Iterable
+        Iterator of GeoJSON records
+    dict
+        Fiona schema of record types
+    """
+    attr_names = [attr.name for attr in scenario.extra_attributes()
+        if attr.type == geom_type.__name__]
```

Comment:
This could be a separate data member in the class instead of relying on class name to have a functional purpose.